frontend/src/pages/AccountSend.vue

```diff
@@ -1067,6 +1067,36 @@ function useSendForm() {
       // Send with Umbra
       const { tx } = await umbra.value.batchSend(signer.value, newSends);
       void txNotify(tx.hash, ethersProvider);
+      if (viewingKeyPair.value?.privateKeyHex && userAddress.value && provider.value) {
+        const batchSendDataPromises = newSends.map(async (item) => {
+          const publicKey = await umbraUtils.lookupRecipient(item.address, ethersProvider, {
+            advanced: false,
+          });
+          return {
+            batch: item,
+            pubKey: publicKey.spendingPublicKey,
+          };
+        });
+        const batchSendData = await Promise.allSettled(batchSendDataPromises).then((values) => {
+          const batches = [];
+          for (const item of values) {
+            if (item.status === 'fulfilled') {
+              batches.push(item.value as { batch: SendBatch; pubKey: string });
+            }
+          }
+          return batches;
+        });
+
+        await batchStoreSend({
```

Comment:
related to the above, wondering if we can simplify/dedupe some code by always using the batch version evne if the length of sends is just 1 (the single send)? maybe the vars we use above are too discrete to enable  this

Reply:
Related https://github.com/ScopeLift/umbra-protocol/pull/524#discussion_r1214469868

Reply:
I removed batch and handled everything in store https://github.com/ScopeLift/umbra-protocol/pull/524/commits/6e68909bd31bc23e5bcb2037a89dd643a8faeb5b